_overviews/scala3-book/control-structures.md

```diff
@@ -989,4 +990,178 @@ finally
 
 Assuming that the `openAndReadAFile` method uses the Java `java.io.*` classes to read a file and doesn't catch its exceptions, attempting to open and read a file can result in both a `FileNotFoundException` and an `IOException`, and those two exceptions are caught in the `catch` block of this example.
 
+## boundary/break
+
+The `boundary`/`break` expression is used to exit a block of code while optionally returning a value. 
+For example, it can be used to break out of a nested loop:
+
+{% tabs control-structures-32 %}
+{% tab 'Scala 3 Only'  %}
+
+```scala
+import scala.util.boundary, boundary.break
+
+val result =
+  boundary:
+    for i <- 1 to 10 do
+      for j <- 1 to 10 do
+        if i == 7 && j == 9 then break("found")
+    "not found"
+
+println(result)
+```
+
+{% endtab %}
+{% endtabs %}
+
+Since `break` uses an exception to control the execution flow, mixing it with `try`/`catch` can lead to unexpected results:
```

Comment:
This one looks like a potential huge footgun if someone doesn't notice `break` throws an exception, so I wanted to mention it. Please notify if the `try`/`catch` example is OK and should stay.

Reply:
`Try` also swallows the control flow because internally it uses `try`/`catch` in a non-`boundary`/`break` compatible way.  It might also be worth explaining how to avoid catching it.  (E.g. `case e: Exception if !e.isInstanceOf[Break[?]]`.)